### ISSUE TITLE
dependency problem with yarn 3 -> graceful JS

### ISSUE THREAD

**Issue by aurelienbottazini** [wallabyjs/public#2979]
### Issue description or question

Hello :)

I am starting a new project with Jest + Typescript

Versions are:
IntelliJIdea2022.1
Node v16.14.2
yarn 3.2.0
Wallaby 1.0.238

Package.json
```
{
  "packageManager": "yarn@3.2.0",
  "devDependencies": {
    "@types/jest": "^27.4.1",
    "@types/x-ray": "^2.3.3",
    "jest": "27",
    "ts-jest": "^27.1.4",
    "typescript": "^4.6.3"
  },
  "dependencies": {
    "x-ray": "^2.3.4"
  }
}
```

.yarnrc.yml

```
yarnPath: .yarn/releases/yarn-3.2.0.cjs

packageExtensions:
  jest-config@*:
    dependencies:
      jest-runtime: "*"
  jest-cli@*:
    dependencies:
      "@jest/reporters": "*"
      "@jest/console": "*"
```

I use the automatic configuration with wallaby

<img width="720" alt="CleanShot 2022-04-28 at 09 42 30@2x" src="https://user-images.githubusercontent.com/32635/165702612-1146996f-6ecf-42aa-a39f-a701c35965e4.png">

Using `yarn jest` the tests run correctly

When I try to start wallaby  I get this error:

```
/Users/aurelienbottazini/projects/wallaby-bug/.pnp.cjs:14867
    throw firstError;
    ^

Error: Your application tried to access graceful-fs, but it isn't declared in your dependencies; this makes the require call ambiguous and unsound.

Required package: graceful-fs
Required by: /Users/aurelienbottazini/Library/Caches/JetBrains/IntelliJIdea2022.1/wallaby/wallaby/

Require stack:
- /Users/aurelienbottazini/Library/Caches/JetBrains/IntelliJIdea2022.1/wallaby/wallaby/server.js
    at Function.require$$0.Module._resolveFilename (/Users/aurelienbottazini/projects/wallaby-bug/.pnp.cjs:14866:13)
    at Function.require$$0.Module._load (/Users/aurelienbottazini/projects/wallaby-bug/.pnp.cjs:14720:42)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at r (/Users/aurelienbottazini/Library/Caches/JetBrains/IntelliJIdea2022.1/wallaby/wallaby/server.js:14:114)
    at /Users/aurelienbottazini/Library/Caches/JetBrains/IntelliJIdea2022.1/wallaby/wallaby/server.js:14:305
    at Object.72../lib/app (/Users/aurelienbottazini/Library/Caches/JetBrains/IntelliJIdea2022.1/wallaby/wallaby/server.js:31:307)
    at r (/Users/aurelienbottazini/Library/Caches/JetBrains/IntelliJIdea2022.1/wallaby/wallaby/server.js:14:254)
    at e (/Users/aurelienbottazini/Library/Caches/JetBrains/IntelliJIdea2022.1/wallaby/wallaby/server.js:14:421)
    at Object.<anonymous> (/Users/aurelienbottazini/Library/Caches/JetBrains/IntelliJIdea2022.1/wallaby/wallaby/server.js:14:438)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47

Process finished with exit code 1
```

I don't understand how to add this dependency to .yarnrcyml
I tried to follow the doc here: https://wallabyjs.com/docs/integration/yarn2.html
I tried various things but there's something I am missing.

I created a minimal reproduction repository here:
https://github.com/aurelienbottazini/wallaby-bug

**Comment by aurelienbottazini** [comment]
When I switch to npm by removing all yarn related files and installing with `npm i` wallaby works.

**Comment by smcenlly** [comment]
Thank you for the sample repo.

We weren't able to reproduce your exact error but we did have an issue. It seems that something changed in PNP's `resolve` patching and Wallaby was crashing internally. Everything is working for us now using your sample repo and the latest version.

Could you please update to Wallaby core `v1.0.1261`, we're hoping it will fix your problem. Wallaby should automatically update it's latest core version if you close and re-open all instances of IntelliJ Idea.

**Comment by aurelienbottazini** [comment]
:( The issue is still there for me:
<img width="1161" alt="CleanShot 2022-04-29 at 15 32 56@2x" src="https://user-images.githubusercontent.com/32635/165954602-0ace70cc-b782-4cc8-ba1f-f2933fa1a54c.png">

**Comment by aurelienbottazini** [comment]
my computer is an m1 macbook air (not sure if that can affect things)

**Comment by smcenlly** [comment]
We are using using M1 Mac and everything is working for us.

Can you please try cloning [your repo](https://github.com/aurelienbottazini/wallaby-bug) to a new directory and see if it works for you?

If it does not, could you please provide updated error details/information?

**Comment by aurelienbottazini** [comment]
Same issue if I clone do a different place.
I am not sure how to provide further information.
The screenshot provided in my earlier message  is the best I can think of regarding details/information. (error is visible with stacktrace + wallabyjs version from the server.js file).

If you can think about something else i am very happy to provide additional information 

fwiw, I switched to npm in the meantime. So I am not blocked.

**Comment by smcenlly** [comment]
We're using all of the same versions of your dependencies and the code you gave us. I can't work out why it's not working for you.

Can you please try and install `graceful-fs` as a dependency of your project (i.e. as a `devDependency`) and see if that fixes the problem?

**Comment by aurelienbottazini** [comment]
Same issue unfortunately.

I am closing the issue as switching to npm works for my use case.

Thanks for taking the time to help!

**Comment by aurelienbottazini** [comment]
Note for future readers: I fixed the issue I had.
It turns out I had different versions of yarn on my computer in different locations. I had all of those locations in my shell `PATH`. This is confusing for yarn. I think it is related to yarn auto discovery features.

What I did:
1. cleaned my `PATH`, i.e making sure I only have one yarn in my path
2. verified I was able to properly use `yarn set version stable`, `yarn set version classic` etc. Previously those commands were not working for me. They did run but It was not reporting the correct version of yarn afterwards when I ran `yarn --version`. This was the symptom that made me understand that my setup was broken.

With my setup _fixed_, I tried the bug repo again and I don't have problems anymore.
